kvserver: TestStoreResolveMetrics is very flaky

`TestStoreResolveMetrics` started flaking very frequently (1 every ~5 runs locally). A bisection points to #98044.

```
=== RUN   TestStoreResolveMetrics
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/09357cecfdbbab5926b4b055936a9b62/logTestStoreResolveMetrics3287525348
    test_log_scope.go:79: use -show-logs to present logs inline
    client_metrics_test.go:251: expected around 200 intent commits, saw 255
    client_metrics_test.go:259: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/09357cecfdbbab5926b4b055936a9b62/logTestStoreResolveMetrics3287525348
--- FAIL: TestStoreResolveMetrics (0.85s)
```

<https://teamcity.cockroachdb.com/project.html?projectId=Cockroach_Ci_TestsGcpLinuxX8664BigVm&testNameId=-2099906381107111862&tab=testDetails>

cc @nvanbenschoten 


Jira issue: CRDB-25248

--- a/pkg/kv/kvserver/client_metrics_test.go
+++ b/pkg/kv/kvserver/client_metrics_test.go
@@ -29,7 +29,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/storage"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
-	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/testcluster"
 	"github.com/cockroachdb/cockroach/pkg/util/hlc"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
@@ -176,8 +175,6 @@ func TestStoreResolveMetrics(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	skip.WithIssue(t, 98404)
-
 	// First prevent rot that would result from adding fields without handling
 	// them everywhere.
 	{
@@ -198,6 +195,12 @@ func TestStoreResolveMetrics(t *testing.T) {
 	key, err := s.ScratchRange()
 	require.NoError(t, err)
 	span := roachpb.Span{Key: key, EndKey: key.Next()}
+
+	// Clear the metrics before starting the test so that we don't count intent
+	// resolutions from server startup.
+	store.Metrics().ResolveCommitCount.Clear()
+	store.Metrics().ResolveAbortCount.Clear()
+	store.Metrics().ResolvePoisonCount.Clear()
 
 	txn := roachpb.MakeTransaction("foo", span.Key, isolation.Serializable, roachpb.MinUserPriority, hlc.Timestamp{WallTime: 123}, 999, int32(s.NodeID()))
 

